nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java

```diff
@@ -904,10 +905,19 @@ public boolean isLoginIdentityProviderEnabled() {
         return !StringUtils.isBlank(getProperty(NiFiProperties.SECURITY_USER_LOGIN_IDENTITY_PROVIDER));
     }
 
+    /**
+     * @return True if property value is 'true'; False otherwise.
+     */
+    public Boolean isAnonymousAuthenticationAllowed() {
+        final String anonymousAuthenticationAllowed = getProperty(SECURITY_ANONYMOUS_AUTHENTICATION, "false");
+
+        return "true".equalsIgnoreCase(anonymousAuthenticationAllowed);
```

Comment:
I think we should trim the retrieved property before comparing. Users frequently have trailing spaces in their `nifi.properties` files, which can cause string inequality when to the human eye it looks valid, and this is difficult to diagnose.

Reply:
I think this is a good idea. However [1] is already filed to address this. Do you want me to update how we treat this property specifically for this PR even if we provide a more general solution later?

[1] https://issues.apache.org/jira/browse/NIFI-7172